src/openforms/appointments/contrib/qmatic/client.py

```diff
@@ -41,3 +82,36 @@ def request(self, method: str, url: str, *args, **kwargs):
             )
 
         return response
+
+    def list_services(self, location_id: str = "") -> list[FullServiceDict]:
+        endpoint = f"branches/{location_id}/services" if location_id else "services"
+        response = self.get(endpoint)
+        response.raise_for_status()
+        return response.json()["serviceList"]
+
+    def list_service_groups(
+        self, service_ids: list[str], location_id: str = ""
+    ) -> list[ServiceGroupDict]:
+        assert service_ids, "Unexpectedly received an empty list of service IDs"
+        params = ";".join(
+            [f"servicePublicId={service_id}" for service_id in service_ids]
+        )
+        endpoint = (
+            (f"v2/branches/{location_id}/services/groups;{params}")
+            if location_id
+            else (f"v1/services/groups;{params}")
+        )
+        response = self.get(endpoint)
+        response.raise_for_status()
+
+        # the shape depends on whether we hit v1 or v2
+        response_data = response.json()
+
+        # v2 API returns a list of service groups (expected to only have one item)
+        if location_id:
+            return response_data
+
+        # v1 API returns a list of branches
+        branches: list[BranchDict] = response_data
+        service_groups = sum((branch["serviceGroups"] for branch in branches), [])
```

Comment:
:) I end up writing the sum over empty list too. I'm always unsure how to write a pythonic flatmap.
Whether it was inside out or outside in... I think it's
`[group for branch in branches for group in branch["serviceGroups"]]`

Reply:
Those sort of nested list comprehensions in python are an abomination